doc/source/introduction.rst

```diff
@@ -1,16 +1,287 @@
+(or, more accurately, any batch of updates) creates a new subdirectory under
+the array directory. Every file in TileDB is **immutable**. TileDB's approach
+prevents **corruption** during a failed update. This means that you do not
+need to nuke your entire (potentially huge) array just because a single
+cell update failed and corrupted your file. Therefore, TileDB offers better
+fault-tolerance and recovery, as well as the opportunity to build extra
+useful features, such as versioning, time-traveling and an "undo" functionality
+for updated.
+
+Handling of Variable-Sized Cell Values
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+TileDB has native support for cell values with variable sizes. In contrast
+to HDF5's VL datatypes, TileDB stores all cell values contiguously in a
+single file, following the same grouping into tiles as the
+fixed-sized values. In addition, it stores the starting offsets of the cell
+values in a separate file. Since the offsets are sorted in ascending order,
+they are highly compressible. This technique leads to both large compression
+ratio and rapid slicing, exhibiting comparable performance to the fixed-sized
+counterparts.
+
+Storage Backend Support
+^^^^^^^^^^^^^^^^^^^^^^^
+
+TileDB is optimized for various storage backends (including HDFS and AWS S3),
+all supported in the core, open-source library. This means that
+(i) this feature is free, and (ii) you get to use it with the embeddable
+TileDB library, without having to set up or communicate with an extra
+service. On the contrary, the HDF5 software does not ship S3 and HDFS
+support by default. The HDF Group offers a cloud service with
+HDFS/S3 integration as part of their
+`Enterprise Support for HDF5 <https://www.hdfgroup.org/solutions/hdf5-enterprise-support-edition/>`_,
+but this is not open-source/free.
+
+Modern Build System
+^^^^^^^^^^^^^^^^^^^
+
+TileDB offers a modern build system that works on Linux, MacOS and Windows,
+with automatic installation of all dependencies. TileDB ships with all
+its compressors by default. This makes it easier to share compressed data,
+since they are all compressed using the same suit of compressors. In contrast,
+HDF5 does not ship with all compressors, but rather offers compressor plug-ins.
+In addition to delegating the installation of the compressors to the users,
+it may lead to a situation where different users use different compressors
+and, hence, cannot read each other's compressed data, unless they install
+the same compressor libraries.
+
+Comparison to SciDB
+-------------------
+
+`SciDB <https://www.paradigm4.com/>`_ is a popular **array
+database management system**. Its functionality, architecture and targeted
+user base are vastly different from those of TileDB (as well as HDF5 that is
+similar to TileDB in those respects). SciDB is a distributed
+database service, not an embeddable library like TileDB. Applications
+interface with SciDB via ODBC connectors, sending queries written in SciDB's
```

Comment:
ODBC is a database thing that science people might not know about, maybe over a socket?